### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by domoritz**
* Convert tests
* Use source maps to map from js to ts

### CODE REVIEW

File: src/compiler/time.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export function formula(timeUnit, fieldRef) {
   return fn + '(' + fieldRef + ')';
 };
 
-export function maxLength(timeUnit, encoding: Encoding) {
+export function maxLength(timeUnit, encoding: any) {
```
Comment (kanitw) [inline src/compiler/time.ts]: why `:any` here?

File: test/compiler/marks.test.ts [hunk 2]
```diff
@@ -1,18 +1,10 @@
-'use strict';
-
 var expect = require('chai').expect;
```
Comment (kanitw) [inline test/compiler/marks.test.ts]: why still require?  (isn’t there a way for node require for external module?)

File: test/compiler/scale.test.ts [hunk 3]
```diff
@@ -30,7 +25,7 @@ describe('vl.compile.scale', function() {
             x: {name: 'x', type: 'O'},
             color: {name: 'color', type: 'O'}
           }
-        }), 'y', 'linear', {}, true);
+        }), 'y', 'linear', {});
```
Comment (kanitw) [inline test/compiler/scale.test.ts]: why changing this?

Comment (domoritz) [inline test/compiler/scale.test.ts]: Because vlscale.domain does not support that many params.

**Comment by kanitw** [comment]
wrong merge target?

File: typings/node.d.ts [hunk 4]
```diff
@@ -524,41 +524,41 @@ declare module "http" {
...
-    /**
-     * Maximum number of sockets to allow per host. Default for Node 0.10 is 5, default for Node 0.12 is Infinity
-     */
-    maxSockets?: number;
-    /**
-     * Maximum number of sockets to leave open in a free state. Only relevant if keepAlive is set to true. Default = 256.
-     */
-    maxFreeSockets?: number;
-  }
+	export interface AgentOptions {
```
Comment (kanitw) [inline typings/node.d.ts]: why this weird tabbing?

**Comment by kanitw** [comment]
Replaced with #718

**Pull request closed without merging**